instrumentation/servlet/src/main/java/brave/servlet/TracingFilter.java

```diff
@@ -0,0 +1,89 @@
+package brave.servlet;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import brave.http.HttpServerParser;
+import brave.propagation.TraceContext;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public final class TracingFilter implements Filter {
```

Comment:
ok thrash is mostly over. Only thing I'm thinking of tuning is how remote addresses are parsed.